src/clj_icat_direct/icat.clj

```diff
@@ -179,7 +178,7 @@
    Returns:
      It returns a sequence of paths."
   [^String user ^String zone ^String folder-path]
-  (map :full_path (run-simple-query :folder-listing user zone folder-path)))
+  (map :full_path (run-simple-query :folder-listing folder-path folder-path user zone)))
```

Comment:
You'll need to test if this works, but I think that if you use `$`-style placeholders in the query, you shouldn't need to pass the path twice. i.e., instead of:

`SELECT foo, bar FROM baz WHERE la = ? and UNION SELECT foo, bar FROM quux WHERE la = ? AND meh = ?`

and doing it as `(run-simple-query :q folder-path folder-path whatever)`, you should be able to do something like:

`SELECT foo, bar FROM baz WHERE la = $1 UNION SELECT foo, bar FROM quux WHERE la = $1 AND meh = $2`

and `(run-simple-query :q folder-path whatever)`

Of course, I could be wrong and the java/clojure thing won't support using that type of placeholder. But I hope it does, it's a nice feature to not have to repeat the same argument multiple times like that.

Reply:
I had the same thought. Unfortunately, I spent some time reading JDBC documentation and `clojure.java.jdbc` unit tests, and I couldn't find any mention of that style of parameter in the documentation for JDBC, `clojure.java.jdbc`, or the Postgres JDBC driver. 😞

Reply:
Ah, it seems that JDBC imposes that restriction and expects you to use `?`. That's annoying.

( https://jdbc.postgresql.org/documentation/head/statement.html )

Reply:
Yeah, it really is. You'd think Oracle would want to fix that.